### ISSUE TITLE
Please add example to godot docs on how to simulate mouse pressed and releases through artifical events

### ISSUE THREAD

**Issue by Shadowblitz16** [godotengine/godot-docs#7515]
**Your Godot version:**
v4.1.beta1.mono.official [828ec2c5d]

**Issue description:**
I am trying to understand how to simulate mouse button presses and releases without moving the mouse with the gamepad and keyboard.

The idea is that I would move the mouse, send a mouse button events and move it back but I can't get the control nodes to recognize the events and I don't know how to pass the correct position

**URL to the documentation page (if already existing):**
na

**Comment by Shadowblitz16** [comment]
> Feel free to use the examples in [godotengine/godot-proposals#6874](https://github.com/godotengine/godot-proposals/issues/6874), I'm not able to write anything up right now but that should be a good start, I'd suggest a comprehensive tutorial rather than specifically this, and different aspects of simulating events from different sources

The works only with custom drawing
I am trying to use the node2d or control's position or global_position property
If I do then it's offseted by some weird value

**Comment by AThousandShips** [comment]
> That works only with custom drawing

I'm talking about my suggestions and code there, nothing about drawing at all

**Comment by Shadowblitz16** [comment]
> > That works only with custom drawing
> 
> I'm talking about my suggestions and code there [godotengine/godot-proposals#6874 (comment)](https://github.com/godotengine/godot-proposals/issues/6874#issuecomment-1554680721), nothing about drawing at all

you assign to mouse_pos instead if position and draw a red circle instead of using a TextureRect

**Comment by AThousandShips** [comment]
Yes, but you can just use the simulated mouse input, you just have to adapt the code a little

But if that's not interesting to you then no need to use it, I just offered the option, my bad

**Comment by Shadowblitz16** [comment]
@AThousandShips controls use different mouse detection.
It's not as simple as just drawing a circle.
I need the engine to pickup that I clicked somewhere that I didn't actually physically click

**Comment by AThousandShips** [comment]
You've confirmed that pushing input events won't work?

**Comment by Shadowblitz16** [comment]
> You've confirmed that pushing input events won't work? I didn't suggest just drawing a circle, I provided code for doing this part, did you test it?

You posted a example that draw a circle but that has nothing to do with my problem.
Yes I have tested it.
My problem is that godot seems to have 20 diffrent spaces for what or where 0,0 is infact I don't think controls and node2d's even use the same space. 

What I am trying to do is convert my node2d position to an event position back to a control position. 

```
func click()->void:
    var evt = InputEventMouseButton.new()
    evt.button_index = MOUSE_BUTTON_LEFT
    evt.position = get_viewport().get_mouse_position() # this does not work it's not in the space space
    evt.pressed = true
    Input.parse_input_event(evt)
    evt.pressed = false
    Input.parse_input_event(evt) 
```

**Comment by AThousandShips** [comment]
I literally posted the instructions to do that, please read my comment :)

**Comment by Shadowblitz16** [comment]
I don't want to confine my mouse because I don't know if the game will be played on desktop or console.
I want to simulate a mouse click at a position the mouse may or may not be at without modifying the mouse.


This is starting to feel like a engine limitation and that https://github.com/godotengine/godot-proposals/issues/6874 should have been implemented

**Comment by AThousandShips** [comment]
Then say that instead of claiming I just showed a circle :)

**Comment by Shadowblitz16** [comment]
Was example 1 not a circle? 
Maybe I got a project mixed up, sorry

I do see that you use a control instead of a node2d I use a animated sprite2d so world space might be diffrent.

**Comment by AThousandShips** [comment]
I gave you two different examples and you reduced them to one instead of presenting your issues with the second one

If you say "just draw a circle" and I give two examples then you're misrepresenting what I've said

**Comment by Shadowblitz16** [comment]
is there a way to do example1 without locking or moving the cursor?

**Comment by AThousandShips** [comment]
The locking is to prevent the mouse from interfeering, as it will relocate the mouse position, and break the sync, can be done without it I think (hard to remember how it worked as I wrote the code some time ago) so feel free to test by removing that part, but might need to compensate for normal mouse input

**Comment by Shadowblitz16** [comment]
would temporary moving it and moving it back work?
would that even be visible in the same frame?

**Comment by AThousandShips** [comment]
I don't know, you play around with it and see!

Yes the motion is requried, and works with the complete example, so I'd suggest some workaround

The reason for the locking of the mouse was to allow co-existing of mouse and controller, and it's possible that further tweaks can allow that even without it, my project was just a starting point to work off of, and unfortunately I don't have the time to play more with it at the moment as I don't need this kind of input in any project I'm working with

**Comment by stephanbogner** [comment]
Adding this information to the docs would be great. Some ideas of what to include:
1. Mentioning things like "Virtual mouse" or "Emulated cursor" as it seems like a common use case
2. Mentioning that mouse-like cursors via gamepads should be avoided as it's less accessible (not sure where I heard it) and [seems less liked](https://www.reddit.com/r/gamedesign/comments/1b5z69v/whats_everyones_thoughts_on_the_mouse_cursor/) (even though games like [Destiny](https://archive.org/details/GDC2016Candland/page/n7/mode/2up) use it)
3. Name-dropping some common games/terms for SEO (Ace Attorney, Monkey Island, point and click adventure, etc.)

**Comment by Shadowblitz16** [comment]
> Adding this information to the docs would be great. Some ideas of what to include:
> 
>     1. Mentioning things like "Virtual mouse" or "Emulated cursor" as it seems like a common use case
> 
>     2. Mentioning that mouse-like cursors via gamepads should be avoided as it's less accessible (not sure where I heard it) and [seems less liked](https://www.reddit.com/r/gamedesign/comments/1b5z69v/whats_everyones_thoughts_on_the_mouse_cursor/) (even though games like [Destiny](https://archive.org/details/GDC2016Candland/page/n7/mode/2up) use it)
> 
>     3. Name-dropping some common games/terms for SEO (Ace Attorney, Monkey Island, point and click adventure, etc.)

Mouse like cursors should not be recommended against just because this engine doesn't support it very well.

**Comment by stephanbogner** [comment]
> > Adding this information to the docs would be great. Some ideas of what to include:
> > ```
> > 1. Mentioning things like "Virtual mouse" or "Emulated cursor" as it seems like a common use case
> > 
> > 2. Mentioning that mouse-like cursors via gamepads should be avoided as it's less accessible (not sure where I heard it) and [seems less liked](https://www.reddit.com/r/gamedesign/comments/1b5z69v/whats_everyones_thoughts_on_the_mouse_cursor/) (even though games like [Destiny](https://archive.org/details/GDC2016Candland/page/n7/mode/2up) use it)
> > 
> > 3. Name-dropping some common games/terms for SEO (Ace Attorney, Monkey Island, point and click adventure, etc.)
> > ```
> 
> Mouse like cursors should not be recommended against just because this engine doesn't support it very well. There are plenty of scenarios that this sort of thing would be a good idea.

Well. I gave two reasons that having nothing to do with engine support.
Of course there are scenarios where they are useful/unavoidable (mentioned in point 3) but generally controlling a cursor with a gamepad – especially for menus – is not that great.

**Comment by stephanbogner** [comment]
In my `cursor.gd`-script the following serves me well (if others find it helpful):
(I'll update my comment as I learn more)

```gdscript
var follow_mouse := false # Whether we use mouse coordinates or gamepad input
var last_tracked_mouse_position := Vector2.ZERO

func _input(event:InputEvent) -> void:
	# Fake mouse clicks and release when on gamepad
	if event.is_action_pressed("select"):
		if not event is InputEventMouseButton:
			trigger_left_mouse_click_at(global_position)
	elif event.is_action_released("select"):
		if not event is InputEventMouseButton:
			trigger_left_mouse_release_at(global_position)

	# Check whether we should use mouse input
	if not follow_mouse and event is InputEventMouseMotion:
		var distance := last_tracked_mouse_position.distance_to(get_global_mouse_position())
		if distance > 5: # This avoids switching with very sensitive mice
			follow_mouse = true
			last_tracked_mouse_position = get_global_mouse_position()

func _process(delta:float) -> void:
	var direction_and_strength := Vector2(Input.get_axis("move_left", "move_right"), Input.get_axis("move_up", "move_down"))

	# Switch to gamepad mode
	if direction_and_strength.length() > 0.5:
		follow_mouse = false

	# Trigger fake mouse move event
	if not follow_mouse:
		trigger_mouse_move_event_at(global_position)

func trigger_left_mouse_click_at(global_coordinates:Vector2) -> void:
	var click_event := InputEventMouseButton.new()
	click_event.device = -1
	click_event.position = global_coordinates
	click_event.button_index = MOUSE_BUTTON_LEFT
	click_event.pressed = true
	Input.parse_input_event(click_event)

func trigger_left_mouse_release_at(global_coordinates:Vector2) -> void:
	var release_event := InputEventMouseButton.new()
	release_event.device = -1
	release_event.position = global_coordinates
	release_event.button_index = MOUSE_BUTTON_LEFT
	release_event.pressed = false
	Input.parse_input_event(release_event)

func trigger_mouse_move_event_at(global_coordinates:Vector2) -> void:
	var move_event := InputEventMouseMotion.new()
	move_event.device = -1
	move_event.position = global_coordinates
	Input.parse_input_event(move_event)
```
